texastribune/donations
# Move circle to subscriptions

#### What's this PR do?
Adds stripe scheduled subscription creation to our /circle membership form. Also uses our new stripe webhooks functionality to listen for subscription creation success events from business membership submission and logs recurring donations and opportunities in Salesforce respectively.

#### Why are we doing this? How does it help us?
This is the next step in a larger modernization project for the donations app that will give us a more reliable and less brittle system for handling recurring donations. This update modernizes the /circle donation form in particular, bringing all of our forms in line with using stripe subscriptions (and scheduled subscriptions, in this case) where applicable.

#### How should this be manually tested?
* Visit the [circle form](https://donations-testing.herokuapp.com/circle) on staging
* Try giving donations for different tiers (i.e. "Editor's", "Founders'", etc.), at different intervals, with or without agreeing to cover fees using a [test card](https://stripe.com/docs/testing#cards)
* For each of these donations...
    * In our stripe dashboard, make sure you are in "Test Mode" (upper corner)
    * Choose the "Customers" tab and you should see the donation you made, listed by the First Name/Last Name you provided
        * <img width="996" alt="Screen Shot 2023-07-21 at 7 52 02 AM" src="https://github.com/texastribune/donations/assets/88053677/14315120-594a-4c33-9217-14063fe9a1eb">
    * Clicking through to the customer you added should show a screen listing a Business subscription
        * <img width="993" alt="Screen Shot 2023-07-21 at 7 54 09 AM" src="https://github.com/texastribune/donations/assets/88053677/e238ba02-0f8a-4e7c-be6a-920341a0603c">
    * Login to our salesforce sandbox (deets listed under "Salesforce Test (2023 refactor)")
    * Follow the instructions [here](https://texastribune.atlassian.net/wiki/spaces/TECH/pages/2273837059/Navigating+Salesforce) to ensure that the donation came through on the salesforce end as expected

#### How should this change be communicated to end users?
We'll let membership team know when this is deployed.

#### Are there any smells or added technical debt to note?
The handling of stripe products is admittedly messy. The hope is that when we refactor the frontend, we can incorporate some more stripe pieces and have the products/prices loaded with the rest of the page instead of being referenced/grabbed here in the midst of the post request.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recwk25CEwmxFf1cm?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
